cli/azd/pkg/account/subscriptions_manager.go

```diff
@@ -173,43 +187,66 @@ func (m *SubscriptionsManager) ListSubscriptions(ctx context.Context) ([]Subscri
 		return nil, fmt.Errorf("listing tenants: %w", err)
 	}
 
-	allSubscriptions := []Subscription{}
-	errors := []error{}
-	oneSuccess := false
-
-	for _, tenant := range tenants {
-		tenantId := *tenant.TenantID
-		subscriptions, err := m.service.ListSubscriptions(ctx, tenantId)
-		if err != nil {
-			errorMsg := err.Error()
-			displayName := *tenant.DisplayName
-			if strings.Contains(errorMsg, "AADSTS50076") {
-				errors = append(
-					errors,
-					fmt.Errorf(
+	span.SetAttributes(fields.AccountSubscriptionsListTenantsFound.Int(len(tenants)))
+
+	listForTenant := func(
+		jobs <-chan armsubscriptions.TenantIDDescription,
+		results chan<- tenantSubsResult,
+		service *azcli.SubscriptionsService) {
+		for tenant := range jobs {
+			azSubs, err := service.ListSubscriptions(ctx, *tenant.TenantID)
+			if err != nil {
+				errorMsg := err.Error()
+				displayName := *tenant.DisplayName
+				if strings.Contains(errorMsg, "AADSTS50076") {
+					err = fmt.Errorf(
 						"%s requires Multi-Factor Authentication (MFA). "+
 							"To authenticate, login with `azd login --tenant-id %s`",
 						displayName,
-						*tenant.DefaultDomain))
-			} else {
-				errors = append(
-					errors,
-					fmt.Errorf("failed to load subscriptions from tenant '%s' : %w", displayName, err))
+						*tenant.DefaultDomain)
+				} else {
+					err = fmt.Errorf("failed to load subscriptions from tenant '%s' : %w", displayName, err)
+				}
 			}
 
+			results <- tenantSubsResult{toSubscriptions(azSubs, *tenant.TenantID), err}
+		}
+	}
+
+	numJobs := len(tenants)
+	jobs := make(chan armsubscriptions.TenantIDDescription, numJobs)
+	results := make(chan tenantSubsResult, numJobs)
+	// Apply max number of concurrent workers at 50
+	numWorkers := int(math.Min(float64(len(tenants)), 50))
+	for i := 0; i < numWorkers; i++ {
+		go listForTenant(jobs, results, m.service)
+	}
+
+	for i := 0; i < numJobs; i++ {
+		jobs <- tenants[i]
+	}
+	close(jobs)
+
+	allSubscriptions := []Subscription{}
+	errors := []error{}
+	oneSuccess := false
+	for i := 0; i < numJobs; i++ {
+		res := <-results
+		if res.err != nil {
+			errors = append(errors, res.err)
 			continue
 		}
 
 		oneSuccess = true
-		for _, subscription := range subscriptions {
-			allSubscriptions = append(allSubscriptions, toSubscription(&subscription, tenantId))
-		}
+		allSubscriptions = append(allSubscriptions, res.subs...)
 	}
 
 	sort.Slice(allSubscriptions, func(i, j int) bool {
 		return allSubscriptions[i].Name < allSubscriptions[j].Name
 	})
 
+	span.SetAttributes(fields.AccountSubscriptionsListTenantsFailed.Int(len(errors)))
```

Comment:
I think we want to `close(results)` after the loop on L242 completes.  It should be safe to do so at that time since there will be no future writers to the channel.

Reply:
@ellismg Just understanding: this is purely from a stylistic standpoint, is that correct? My understanding is that the channel would get GC'd the same way. I'm not sure if `close` actually results in any immediate memory changes.

Reply:
That's my understanding as well, @weikanglim.